pkg/pyramid/file_locker.go

```diff
@@ -0,0 +1,44 @@
+package pyramid
+
+import (
+	"sync"
+
+	"github.com/treeverse/lakefs/pkg/pyramid/params"
+)
+
+// fileLocker allows file locking to avoid race conditions between cache rejection/eviction
+// and the file usage by TierFS
+type fileLocker struct {
+	wgMap *sync.Map
```

Comment:
1. Consider using map+mutex (usually it will work faster)
2. Using a map with mutex will remove the extra allocation of waitgroup in case there is already a waitgroup on the path